proto/substrait/algebra.proto

```diff
@@ -881,7 +882,21 @@ message Expression {
     message IntervalDayToSecond {
       int32 days = 1;
       int32 seconds = 2;
-      int32 microseconds = 3;
+
+      // oneof ensures that a user can only use precision (and subseconds) or microseconds
```

Comment:
I don't think we need the comment (a reader of protobufs should know how oneof works).

Reply:
moved things around a little. the pairing of precision with subseconds is the part that is a little unusual with oneof.